chainerio/filesystems/hdfs.py

```diff
@@ -222,7 +222,26 @@ def _recursive_list(self, full_uri_prefix_offset, path):
 
     def stat(self, path):
         self._create_connection()
-        return self.connection.stat(path)
+        info = self.connection.info(path)
+        kind = info['kind']
+        mode = info['permissions']
+        if kind == 'file':
+            mode |= 0x8000          # _S_IFREG
+        elif kind == 'directory':
+            mode |= 0x4000          # _S_IFDIR
+
+        return os.stat_result([
+            mode,                   # st_mode
+            None,                   # st_ino
+            None,                   # st_dev
+            None,                   # st_nlink
+            info['owner'],          # st_uid
```

Comment:
I think the `info['owner']` and `info['group']` are strings instead of uids

Reply:
Yes they are! In contrast what `os.stat` returns are int (uid/gid). It's fortunate (?) that `os.stat_result` also accepts strings, but if a user expects a normal behavior (having int), it'll fail.

```
In [4]: os.stat('pytest.ini')       
Out[4]: os.stat_result(st_mode=33204, st_ino=27665508, st_dev=2049, st_nlink=1, st_uid=1000, st_gid=1000, st_size=51, st_atime=1585040937, st_mtime=1583388582, st_ctime=1583388582)
```

Reply:
Oh, I did not know `os.stat_result` also accepts strings, I thought it just a wrapper of `struct stat`. There is actually a way to convert the username to uid, and I used that when writing the hdfs fuse client. However, we still have the problem when the user or group does not exist on the local machine. 
https://stackoverflow.com/questions/826082/python-finding-uid-gid-for-a-given-username-groupname-for-os-chown